src/aiida_quantumespresso/data/hubbard_structure.py

```diff
@@ -0,0 +1,225 @@
+        :param neighbour_index: neighbouring atom name in unitcell
+        :param neighbour_manifold: neighbour manifold (e.g. 3d, 3d-2p)
+        :param value: value of the Hubbard parameter, in eV
+        :param hubbard_type: hubbard type (U, V, J, ...), defaults to 'V'
+            (see :meth:`~aiida_quantumespresso.common.hubbard.Hubbard` for full allowed values)
+        :param use_kinds: whether to use kinds for initializing the parameters; when False, it
+            initializes all the ``Kinds`` matching the ``atom_name``
+        """
+        sites = self.get_pymatgen_structure().sites
+
+        function = self._get_one_kind_index if use_kinds else self._get_symbol_indecis
+        atom_indecis = function(atom_name)
+        neigh_indecis = function(neighbour_name)
+
+        if atom_indecis is None or neigh_indecis is None:
+            raise ValueError('species or kind names not in structure')
+
+        for atom_index in atom_indecis:
+            for neighbour_index in neigh_indecis:
+                _, translation = sites[atom_index].distance_and_image(sites[neighbour_index])
+                translation = np.array(translation, dtype=np.int64).tolist()
+                args = (
+                    atom_index, atom_manifold, neighbour_index, neighbour_manifold, value, translation, hubbard_type
+                )
+                self.append_hubbard_parameter(*args)
+
+    def initialize_onsites_hubbard(
+        self,
+        atom_name: str,
+        atom_manifold: str,
+        value: float = 1e-8,
+        hubbard_type: str = 'Ueff',
+        use_kinds: bool = True,
+    ):
+        """Initialize and append onsite Hubbard values of atoms with specific name.
+
+        :param atom_name: atom name in unitcell
+        :param atom_manifold: atomic manifold (e.g. 3d, 3d-2p)
+        :param value: value of the Hubbard parameter, in eV
+        :param hubbard_type: hubbard type (U, J, ...), defaults to 'Ueff'
+            (see :meth:`~aiida_quantumespresso.common.hubbard.Hubbard` for full allowed values)
+        :param use_kinds: whether to use kinds for initializing the parameters; when False, it
+            initializes all the ``Kinds`` matching the ``atom_name``
+        """
+        function = self._get_one_kind_index if use_kinds else self._get_symbol_indecis
+        atom_indecis = function(atom_name)
+
+        if atom_indecis is None:
+            raise ValueError('species or kind names not in structure')
+
+        for atom_index in atom_indecis:
+            args = (atom_index, atom_manifold, atom_index, atom_manifold, value, [0, 0, 0], hubbard_type)
+            self.append_hubbard_parameter(*args)
+
+    def _get_one_kind_index(self, kind_name: str) -> List[int]:
+        """Return the first site index matching with `kind_name`."""
+        for i, site in enumerate(self.sites):
+            if site.kind_name == kind_name:
+                return [i]
+
+    def _get_symbol_indecis(self, symbol: str) -> List[int]:
```

Comment:
If with `indecis` you mean the plural of `index` it is `indices`
```suggestion
    def _get_symbol_indices(self, symbol: str) -> List[int]:
```